src/backend/distributed/commands/table.c

```diff
@@ -654,6 +657,14 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 	 */
 	bool deparseAT = false;
 	bool propagateCommandToWorkers = true;
+
+	/*
+	 * Sometimes we want to run a different DDL Command string in MX workers
+	 * For example, in cases where worker_nextval should be used instead
+	 * of nextval() in column defaults with type int and smallint
+	 */
+	bool useInitialDDLCommandString = true;
```

Comment:
One the one hand this is a bit ugly, but on the other hand, the root of the ugliness is that we use the original command string at all, so I'm happy we're making that explicit.